### DISCUSSION TITLE
Enable HTTPS for self hosted instance?

### DISCUSSION THREAD

**Discussion by SechyMonkey** [mawoka-myblock/ClassQuiz#111] [Q&A] [answered]
Hello!

I'm new to running things in containers and have also never used Caddy or Gunicorn before. I've been attempting to setup everything for self-hosting on a local test machine I have in my home lab but have been unsuccessful in getting HTTPS/SSL setup and working properly. 

Are there any tips on how I might be able to get this running? I've worked with looking up Caddy documentation and tried changing around a few things and adding in my certificates to the caddyfile for my testing domain without success. Whenever I try to connect on any port other than :8000 I get refused connection. And I temporarily enabled google auth to work on http only and got the login to work, but I'd love to get this functional for https.

Any advice or information would be greatly appreciated.

**Comment by mawoka-myblock** [comment, marked as answer]
The file you want to change is the following: https://github.com/mawoka-myblock/ClassQuiz/blob/master/Caddyfile-docker
 You can replace the `:8080` with your domain and Caddy will automatically get an https-certificate for you.

If you already have nginx or apache running, you can put ClassQuiz behind it, so you don't have to change anything regarding Caddy. That's what I also do at classquiz.de.

**Reply by SechyMonkey** [reply]
Thanks for the information! I'll clearly have to do a lot more reading regarding docker containers and caddy in general.

**Reply by mawoka-myblock** [reply]
If you have any questions (also regarding docker), feel free to ask me! Maybe not here, but via matrix, email, etc: https://mawoka.eu/contact

**Reply by TheNomad11** [reply]
Do you have an example of Nginx or Apache reverse proxy @mawoka-myblock ?

**Reply by mawoka-myblock** [reply]
Only for nginx. But it's just the normal standard reverse proxy. No special headers

**Reply by TheNomad11** [reply]
so no sockets or websockets etc?

**Reply by TheNomad11** [reply]
this was an issue that made me wonder https://github.com/mawoka-myblock/ClassQuiz/issues/252#issuecomment-1607051955

**Reply by mawoka-myblock** [reply]
That's the config for classquiz.de:

```
server {
    server_name classquiz.de;
    listen 443 ssl http2;

    # Proxy Config
    location / {
        proxy_pass http://localhost:8044;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	client_max_body_size 200M;
    }



    ssl_certificate fullchain.pem; # managed by Certbot
    ssl_certificate_key privkey.pem; # managed by Certbot
    include options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam ssl-dhparams.pem; # managed by Certbot



}

server {
    return 301 https://classquiz.de$request_uri;


    server_name www.classquiz.de;


    listen 80;
    listen 443 ssl;
    ssl_certificate ullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/classquiz.de/privkey.pem;
    include ptions-ssl-nginx.conf;
    ssl_dhparam ssl-dhparams.pem;

}

server {
    if ($host = classquiz.de) {
        return 301 https://$host$request_uri;
    }
    # managed by Certbot


    server_name classquiz.de;
    listen 80;
    return 404; # managed by Certbot


}
```

**Reply by TheNomad11** [reply]
thanks a lot! might be a good idea  to add it to the documentation?
